modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/AbstractCreateIndexCommandParams.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.catalog.commands;
+
+import java.util.List;
+
+/** Abstract create index ddl command. */
+public abstract class AbstractCreateIndexCommandParams extends AbstractIndexCommandParams {
+    /** Table name. */
+    protected String tableName;
+
+    /** Unique index flag. */
+    protected boolean unique;
+
+    /** Indexed columns. */
+    protected List<String> columns;
+
+    /** Returns table name. */
+    public String tableName() {
+        return tableName;
+    }
+
+    /** Returns {@code true} if index is unique, {@code false} otherwise. */
+    public boolean unique() {
+        return unique;
+    }
+
+    /** Returns index columns. */
+    public List<String> columns() {
+        return columns;
+    }
+
+    /** Parameters builder. */
+    protected abstract static class AbstractCreateIndexBuilder<ParamT extends AbstractCreateIndexCommandParams, BuilderT> extends
```

Comment:
I think this rule make sense in case of a single parameter or in case of general interface, where parameters have no particular meaning like `java.util.Function` (when <A, B, C> looks less irritating to the eyes than <Arg1T, Arg2T,...>).
Here long parameter names looks fine.

Reply:
Neighboring classes already use this naming, I wouldn't want to change that.

`org.apache.ignite.internal.catalog.commands.AbstractTableCommandParams.AbstractBuilder`
`org.apache.ignite.internal.catalog.commands.AbstractZoneCommandParams.AbstractBuilder`